reya-rust-sdk/src/http_provider.rs

```diff
@@ -546,12 +548,75 @@ impl HttpProvider {
                 return Some(transaction_receipt.clone()?);
             }
             Err(err) => {
-                error!("{:?}", err);
+                error!(
+                    "Failed to get tx receipt hash {:?} with error {:?}",
+                    tx_hash, err
+                );
                 return None;
             }
         }
     }
 
+    async fn poll_for_receipt(
+        &self,
+        tx_hash: alloy_primitives::FixedBytes<32>,
+    ) -> Option<rpc::types::TransactionReceipt> {
+        let polling_interval = Duration::from_secs(1);
+        let mut run = 0;
+
+        loop {
+            match self.get_transaction_receipt(tx_hash).await {
```

Comment:
hmm, this polling for receipt logic is quite clanky, why don't we just await it when submitting tx?

Reply:
Yes that would be ideal but I monitored the waiting time for the confirmation and it was very slow (7 seconds). So I think it's quicker to poll at the end, after we sent al txs. By then the 1st ones would have already been confirmed

Reply:
Ideally, we'd send this load to another process